### PR TITLE
SQOOP-3423 Let user pass password to connect Hive when it set to LDAP…

### PR DESCRIPTION
**Pull request by dionusos**
… authentication

### CODE REVIEW

File: src/java/org/apache/sqoop/hive/HiveServer2ConnectionFactoryInitializer.java [hunk 1]
```diff
@@ -53,6 +57,13 @@ private String determineConnectionUsername(SqoopOptions sqoopOptions) {
     }
   }
 
+  private String determineConnectionPassword(SqoopOptions sqoopOptions) {
```
Comment (szvasas) [inline src/java/org/apache/sqoop/hive/HiveServer2ConnectionFactoryInitializer.java]: Is the empty string ("") a legal password in LDAP environment?
If yes then we could eliminate this method and use sqoopOptions.getHs2Password() instead.

**Review by szvasas — CHANGES_REQUESTED** [review]
Hi @dionusos,

Thank you for the patch, it looks great, I have found only 2 minor things which need to be addressed before merging:
- hive.txt has also some documentation about the HiveServer2 connection("Currently only Kerberos authentication and text file format is supported with this option."), please make sure that it mentions the LDAP authentication too (

File: src/java/org/apache/sqoop/hive/HiveServer2ConnectionFactoryInitializer.java (hunk 1 above)
Comment (dionusos) [reply src/java/org/apache/sqoop/hive/HiveServer2ConnectionFactoryInitializer.java]: As I found in LDAP environment empty password is possible so as the recommendation I removed the extra method.

**Review by szvasas — APPROVED** [review]

**Comment by ebogi** [comment]
LGTM, ship it!

**Pull request merged**